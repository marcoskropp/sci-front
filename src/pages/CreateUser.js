import React, { Component } from "react";
import { Link } from "react-router-dom";
import {
  Button,
  Container,
  FormControl,
  Grid,
  InputLabel,
  MenuItem,
  TextField,
  Select,
  Typography
} from "@material-ui/core";

export default class CreateUser extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: "",
      username: "",
      email: "",
      password: "",
      role_id: 1,
      roles: [{ id: 1, name: "Professor" }, { id: 2, name: "Aluno" }]
    };

    this.link = React.forwardRef((props, ref) => (
      <Link innerRef={ref} {...props} />
    ));
  }

  handleInputChange = event => {
    console.log(event);

    this.setState({ [event.target.name]: event.target.value });
  };

  handleSubmit = async event => {
    event.preventDefault();
  };

  render() {
    return (
      <Container>
        <div>
          <Typography variant="h4" component="h2" gutterBottom>
            REGISTRAR
          </Typography>
          {this.renderForm()}
        </div>
      </Container>
    );
  }

  renderForm = () => (
    <form onSubmit={this.handleSubmit}>
      <Grid container spacing={3} justify="center">
        <Grid item xs={12}>
          <TextField
            fullWidth
            label="Nome"
            name="name"
            value={this.state.name}
            variant="outlined"
            onChange={this.handleInputChange}
          />
        </Grid>
        <Grid item xs={6}>
          <TextField
            fullWidth
            label="Username"
            name="username"
            value={this.state.username}
            variant="outlined"
            onChange={this.handleInputChange}
          />
        </Grid>
        <Grid item xs={6}>
          <TextField
            fullWidth
            label="Email"
            name="email"
            onChange={this.handleInputChange}
            value={this.state.email}
            variant="outlined"
          />
        </Grid>
        <Grid item xs={6}>
          <TextField
            fullWidth
            label="Password"
            name="password"
            onChange={this.handleInputChange}
            value={this.state.password}
            variant="outlined"
            type="password"
          />
        </Grid>
        <Grid item xs={6}>
          <FormControl variant="outlined" fullWidth>
            <InputLabel variant="outlined" htmlFor="role_id">
              Tipo
            </InputLabel>
            <Select
              name="role_id"
              onChange={this.handleInputChange}
              value={this.state.role_id}
              labelWidth={35}
              inputProps={{ id: "role_id", name: "role_id" }}
            >
              {this.renderOptions()}
            </Select>
          </FormControl>
          {/* <select name="role_id">{this.renderOptions()}</select> */}
        </Grid>
        <Grid item xs={6}>
          <Button type="submit" variant="contained" color="primary">
            Cadastrar
          </Button>
        </Grid>
        <Grid item xs={6}>
          <Button
            variant="contained"
            color="secondary"
            to="/"
            component={this.link}
          >
            Voltar
          </Button>
        </Grid>
      </Grid>
    </form>
  );

  renderOptions = () =>
    // this.state.roles.map(role => <option value={role.id}>{role.name}</option>);
    this.state.roles.map(role => (
      <MenuItem value={role.id}>{role.name}</MenuItem>
    ));
}
