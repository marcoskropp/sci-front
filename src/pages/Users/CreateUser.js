import React, { useState, useEffect, forwardRef } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import {
  Button,
  Container,
  FormControl,
  Grid,
  InputLabel,
  MenuItem,
  Paper,
  Select,
  TextField,
  Typography
} from '@material-ui/core';

import { index } from '../../services/Roles.services';
import { store } from '../../services/Users.services';

export default function CreateUser(props) {
  const [user, setUser] = useState({
    name: '',
    username: '',
    email: '',
    password: '',
    roleId: ''
  });
  const [roles, setRoles] = useState([]);

  const link = forwardRef((props, ref) => <Link innerRef={ref} {...props} />);

  useEffect(() => {
    const getRoles = async () => {
      const roles = await index();
      setRoles(roles);
      setUser(user => ({ ...user, roleId: roles[0].id }));
    };
    getRoles();
  }, []);

  const handleSubmit = async event => {
    event.preventDefault();

    const { roleId, ...rest } = user;

    await store({ ...rest, role_id: roleId });

    props.history.push('/');
  };

  const handleChange = event => {
    const { name, value } = event.target;

    setUser({ ...user, [name]: value });
  };

  const renderOptions = () => roles.map(role => {
    if (user.roleId === role.id) {
      return (
        <MenuItem key={role.id} value={role.id} selected>
          {role.name}
        </MenuItem>
      );
    }
    return (
      <MenuItem key={role.id} value={role.id}>
        {role.name}
      </MenuItem>
    );
  });

  const renderForm = () => {
    const {
      name, username, email, password, roleId
    } = user;

    return (
      <form onSubmit={handleSubmit}>
        <Grid container spacing={3} justify="center" alignItems="center">
          <Grid item xs={12}>
            <TextField
              fullWidth
              label="Nome"
              name="name"
              value={name}
              variant="outlined"
              onChange={handleChange}
            />
          </Grid>
          <Grid item xs={12} md={6}>
            <TextField
              fullWidth
              label="Username"
              name="username"
              value={username}
              variant="outlined"
              onChange={handleChange}
            />
          </Grid>
          <Grid item xs={12} md={6}>
            <TextField
              fullWidth
              label="Email"
              name="email"
              onChange={handleChange}
              value={email}
              variant="outlined"
            />
          </Grid>
          <Grid item xs={12} md={6}>
            <TextField
              fullWidth
              label="Password"
              name="password"
              onChange={handleChange}
              value={password}
              variant="outlined"
              type="password"
            />
          </Grid>
          <Grid item xs={12} md={6}>
            <FormControl variant="outlined" fullWidth>
              <InputLabel variant="outlined" htmlFor="roleId">
                Tipo
              </InputLabel>
              <Select
                name="roleId"
                onChange={handleChange}
                value={roleId}
                labelWidth={35}
                inputProps={{ id: 'roleId', name: 'roleId' }}
              >
                {renderOptions()}
              </Select>
            </FormControl>
          </Grid>
          <Grid item xs={12}>
            <Button type="submit" variant="contained" color="primary">
              Cadastrar
            </Button>
            &nbsp;
            <Button
              variant="contained"
              color="secondary"
              to="/"
              component={link}
            >
              Voltar
            </Button>
          </Grid>
        </Grid>
      </form>
    );
  };

  return (
    <Container>
      <br />
      <Paper style={{ padding: '30px' }}>
        <Typography variant="h4" component="h2" gutterBottom>
          REGISTRAR
        </Typography>
        {renderForm()}
      </Paper>
    </Container>
  );
}

CreateUser.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func
  })
};
