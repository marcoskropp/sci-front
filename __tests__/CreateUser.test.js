import React from 'react';
import { shallow } from 'enzyme';

import CreateUser from '../src/pages/Users/CreateUser';

describe('Testing CreateUser Page', () => {
  it('Snapshot renders', () => {
    const wrapper = shallow(<CreateUser />);

    expect(wrapper).toMatchSnapshot();
  });
});
