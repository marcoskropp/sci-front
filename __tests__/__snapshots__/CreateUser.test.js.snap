// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CreateUser Page Snapshot renders 1`] = `ShallowWrapper {}`;
